### ISSUE TITLE
Assertion failure in GAP master branch

### ISSUE THREAD

**Issue by alex-konovalov** [simpcomp-team/simpcomp#7]
As you can see in the log file from https://travis-ci.org/gap-system/gap-docker-pkg-tests-master/jobs/357635614 there is an assertion failure 
```
########> Diff in /home/gap/inst/gap-master/pkg/simpcomp/tst/simpcomp.tst:1183
# Input is:
ll:=SCsFromGroupExt(G,16,4,0,0,false,false,0,[]);;
# Expected output:
# But found:
Error, Assertion failure
########
```
It does not happen when no packages are loaded (GAP started with -r option), but happens in these two cases:
- With all packages loaded with LoadAllPackages()    
- with default packages loaded at GAP startup  

It could be that the problem is not in SimpComp but in some other package, but this has to be investigated.

**Comment by simpcomp-team** [comment]
Is there any way for us to reproduce this error?

**Comment by alex-konovalov** [comment]
In the master branch of GAP, start GAP with default packages and call  

    TestPackage("simpcomp");

This is what I have then:
```
$ bin/gap.sh -r 
 ┌───────┐   GAP 4.10dev-714-gd6d1b20 of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin16.7.0-default64
 Configuration:  gmp 6.1.2, readline
 Loading the library and packages ...
 Packages:   AClib 1.3, Alnuth 3.1.0, AtlasRep 1.5.1, AutPGrp 1.9, Browse 1.8.7, CRISP 1.4.4, Cryst 4.1.17, CrystCat 1.1.8, 
             CTblLib 1.2.2, FactInt 1.6.2, FGA 1.4.0, GAPDoc 1.6.1, IO 4.5.1, IRREDSOL 1.4, LAGUNA 3.9.0, Polenta 1.3.8, 
             Polycyclic 2.13.1, PrimGrp 3.3.1, RadiRoot 2.8, ResClasses 4.7.1, SmallGrp 1.3, Sophus 1.24, SpinSym 1.5, 
             TomLib 1.2.6, TransGrp 2.0.2, utils 0.54
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> TestPackage("simpcomp");
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  AutoDoc 2018.02.14 (Generate documentation from GAP source code)
by Sebastian Gutsche (http://wwwb.math.rwth-aachen.de/~gutsche/) and
   Max Horn (http://www.quendi.de/math).
Homepage: https://gap-packages.github.io/AutoDoc
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  Gauss 2018.04.03 (Extended Gauss functionality for GAP)
by Simon Goertzen (http://wwwb.math.rwth-aachen.de/goertzen).
Homepage: http://homalg-project.github.io/homalg_project/Gauss/
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  ToolsForHomalg 2018.01.30 (Provides special methods and knowledge propagation tools)
by Mohamed Barakat (http://www.mathematik.uni-kl.de/~barakat/),
   Sebastian Gutsche (http://wwwb.math.rwth-aachen.de/~gutsche/), and
   Markus Lange-Hegermann (http://wwwb.math.rwth-aachen.de/~markus/).
Homepage: http://homalg-project.github.io/homalg_project/ToolsForHomalg/
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  homalg 2017.10.26 (A homological algebra meta-package for computable Abelian categories)
by Mohamed Barakat (http://www.mathematik.uni-kl.de/~barakat/),
   Sebastian Gutsche (http://wwwb.math.rwth-aachen.de/~gutsche/), and
   Markus Lange-Hegermann (http://wwwb.math.rwth-aachen.de/~markus/).
Homepage: http://homalg-project.github.io/homalg_project/homalg/
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  MatricesForHomalg 2018.04.06 (Lazy evaluated matrices with clever operations for the homalg project)
by Mohamed Barakat (http://www.mathematik.uni-kl.de/~barakat/),
   Markus Lange-Hegermann (http://wwwb.math.rwth-aachen.de/~markus/),
   Martin Leuner (http://wwwb.math.rwth-aachen.de/Mitarbeiter/leuner.php), and
   Vinay Wagh (http://www.iitg.ernet.in/vinay.wagh/).
Homepage: http://homalg-project.github.io/homalg_project/MatricesForHomalg/
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  GaussForHomalg 2018.02.05 (Gauss functionality for the homalg project)
by Simon Goertzen (http://wwwb.math.rwth-aachen.de/goertzen).
Homepage: http://homalg-project.github.io/homalg_project/GaussForHomalg/
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  Modules 2018.03.20 (A homalg based package for the Abelian category of finitely presented modules over computable rings)
by Thomas Bächler (http://wwwb.math.rwth-aachen.de/~thomas/),
   Mohamed Barakat (http://www.mathematik.uni-kl.de/~barakat/),
   Florian Diebold (diebold@mathematik.uni-kl.de),
   Sebastian Gutsche (http://wwwb.math.rwth-aachen.de/~gutsche/),
   Markus Lange-Hegermann (http://wwwb.math.rwth-aachen.de/~markus/), and
   Vinay Wagh (http://www.iitg.ernet.in/vinay.wagh/).
Homepage: http://homalg-project.github.io/homalg_project/Modules/
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  GRAPE 4.7 (GRaph Algorithms using PErmutation groups)
by Leonard H. Soicher (http://www.maths.qmul.ac.uk/~leonard/).
Homepage: http://www.maths.qmul.ac.uk/~leonard/grape/
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
#I  simpcomp: package `homology' not installed or its binaries are not available, falling back to (slower) internal homology algorithms.
Loading simpcomp 2.1.8
by F. Effenberger and J. Spreer
https://github.com/simpcomp-team/simpcomp
########> Diff in /Users/alexk/GITREPS/gap/pkg/simpcomp/tst/simpcomp.tst:1183
# Input is:
ll:=SCsFromGroupExt(G,16,4,0,0,false,false,0,[]);;
# Expected output:
# But found:
Error, Assertion failure
########
########> Diff in /Users/alexk/GITREPS/gap/pkg/simpcomp/tst/simpcomp.tst:1184
# Input is:
Size(ll);
# Expected output:
4
# But found:
Error, Variable: 'll' must have a value
########
########> Diff in /Users/alexk/GITREPS/gap/pkg/simpcomp/tst/simpcomp.tst:1186
# Input is:
SCIsIsomorphic(ll[1],K3);
# Expected output:
true
# But found:
Error, Variable: 'll' must have a value
########
simpcomp package test
msecs: 5868
#I  Errors detected while testing package simpcomp version 2.1.8
#I  using the test file `/Users/alexk/GITREPS/gap/pkg/simpcomp/tst/simpcomp.tst'
gap> 

```

**Comment by simpcomp-team** [comment]
With GAP 4.9.1 this error does not occur. 

GAP 4.10dev-714-gd6d1b20 does not start on my machine for some reason.

**Comment by alex-konovalov** [comment]
Yes, as I said, you need master branch of GAP for that. What exactly happens when you try to start it?

**Comment by simpcomp-team** [comment]
It says

```jspreer@smp-c8p3tz1:~/apps/gap/bin$ ./gap.sh
 ┌───────┐   GAP 4.10dev-714-gd6d1b20 of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64
 Configuration:  gmp 6.1.0, readline
 Loading the library and packages ...
Error, LOAD_DYN: kernel module built for older version of GAP in
  if not LOAD_DYN( arg[1], false ) then
    Error( "no support for dynamic loading" );
fi; at /home/jspreer/apps/gap/lib/files.gd:583 called from 
<function "LoadDynamicModule">( <arguments> )
 called from read-eval loop at /home/jspreer/apps/gap/pkg/io-4.5.1/init.g:23
type 'quit;' to quit to outer loop
brk>
```

(sorry for the late answer, thought I sent this days ago...)

**Comment by alex-konovalov** [comment]
The error message is for the kernel module from the IO package. You need to rebuild package(s) after rebuilding the kernel.

In practice, usually I do this by doing  
```
rm -r pkg
make bootstrap-pkg-full
cd pkg
../bin/BuildPackages.sh
```
It takes longer time than rebuilding one package, but does all in once and does not require a lot of  hand input from me. I do not keep development versions of my packages in the main GAP root (they are in `.gap/pkg` instead), so can easily delete `pkg` subdirectory.

**Comment by simpcomp-team** [comment]
Sorry (again), just realized I never replied. 

Everything works now, thanks. Will now start working on the diffs.

**Comment by simpcomp-team** [comment]
This issue can be merged with #10
